### PR TITLE
Expose connecting player's UUID in the PreLoginEvent

### PR DESCRIPTION
**Pull request by kezz**
Unsure if docs need to be more specific, or if the old event constructor should be deprecated/removed.

### CODE REVIEW

**Comment by 0utplay** [comment]
I think using `uniqueId` as field, parameter and getter name is more appropriate as the API usally uses that for any uuids.

https://github.com/PaperMC/Velocity/blob/7f776abf550001186a1f11f4e5ff760dd66e3b04/api/src/main/java/com/velocitypowered/api/proxy/Player.java#L75

File: proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/ServerLogin.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import com.velocitypowered.proxy.protocol.ProtocolUtils.Direction;
 import com.velocitypowered.proxy.util.except.QuietDecoderException;
 import io.netty.buffer.ByteBuf;
+import java.util.Objects;
```
Comment (astei) [inline proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/ServerLogin.java]: Unused import

File: api/src/main/java/com/velocitypowered/api/event/connection/PreLoginEvent.java [hunk 2]
```diff
@@ -33,17 +34,30 @@ public final class PreLoginEvent implements ResultedEvent<PreLoginEvent.PreLogin
...
   /**
-   * Creates a new instance.
+   * Creates a new instance, without an associated UUID.
    *
    * @param connection the connection logging into the proxy
    * @param username the player's username
    */
   public PreLoginEvent(InboundConnection connection, String username) {
+    this(connection, username, null);
+  }
```
Comment (astei) [inline api/src/main/java/com/velocitypowered/api/event/connection/PreLoginEvent.java]: let's deprecate this

**Review by astei — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/velocitypowered/api/event/connection/PreLoginEvent.java [hunk 3]
```diff
@@ -55,6 +69,15 @@ public String getUsername() {
     return username;
   }
 
+  /**
+   * Returns the UUID of the connecting player. This value is {@code null} on 1.19.1 and lower.
+   *
+   * @return the uuid
+   */
+  public @Nullable UUID getUuid() {
```
Comment (astei) [inline api/src/main/java/com/velocitypowered/api/event/connection/PreLoginEvent.java]: Change this to `getUniqueId()`

**Review by astei — CHANGES_REQUESTED** [review]

**Comment by kyngs** [comment]
Isn't the GameProfileRequestEvent called after the PreLoginEvent? If yes, what's the point of exposing the UUID even though it still cannot be used for DB purposes?

File: proxy/src/main/java/com/velocitypowered/proxy/connection/client/InitialLoginSessionHandler.java [hunk 4]
```diff
@@ -119,7 +119,7 @@ public boolean handle(ServerLogin packet) {
     inbound.setPlayerKey(playerKey);
     this.login = packet;
 
-    PreLoginEvent event = new PreLoginEvent(inbound, login.getUsername());
+    PreLoginEvent event = new PreLoginEvent(inbound, login.getUsername(), login.getHolderUuid());
```
Comment (456dev) [inline proxy/src/main/java/com/velocitypowered/proxy/connection/client/InitialLoginSessionHandler.java]: is this the uuid sent by the client?

Comment (4drian3d) [reply proxy/src/main/java/com/velocitypowered/proxy/connection/client/InitialLoginSessionHandler.java]: Yes

**Review by 4drian3d — APPROVED** [review]

**Pull request merged**